ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -292,4 +348,66 @@ protected String getReplicaThreadName(String datacenterToReplicateFrom, int thre
   protected boolean shouldReplicateFromDc(String datacenterName) {
     return cloudConfig.vcrSourceDatacenters.contains(datacenterName);
   }
+
+  /**
+   * Check and schedule a VCR helix update task. We schedule updateVcrHelix in 2 cases:
+   * 1. On a node become online role of partition, which usually happens on restart or deployment.
+   * 2. On Ambry cluster change.
+   */
+  private void scheduleVcrHelix() {
+    if (vcrHelixUpdateFuture != null && vcrHelixUpdateFuture.cancel(false)) {
+      // If a vcrHelixUpdate task is scheduled, try to cancel it first.
+      logger.info("There was a scheduled vcrHelixUpdate task. Canceled.");
+    }
+    // either success cancel or not, we should schedule a new job to updateVcrHelix
+    vcrHelixUpdateFuture =
+        scheduler.schedule(() -> updateVcrHelix(), cloudConfig.vcrHelixUpdateDelayTimeInSeconds, TimeUnit.SECONDS);
+    logger.info("VcrHelixUpdate task scheduled. Will run in {} seconds.", cloudConfig.vcrHelixUpdateDelayTimeInSeconds);
+  }
+
+  /**
+   * The actual performer to update VCR Helix:
+   */
+  private void updateVcrHelix() {
+    String localDcZkStr = ((HelixClusterManager) clusterMap).getLocalDcZkConnectString();
```

Comment:
I think its best to make the method `updateVcrHelix` synchronized. There are race conditions that might cause two threads to run this method simultaneously. For example, `vcrHelixUpdateFuture.cancel` method in `scheduleVcrHelix` will not cancel an already running thread. In that case two threads might try to `updateVcrHelix`.
Note that this method is not a performance critical method anyway, and we are already waiting for sometime before scheduling it to avoid a burst of calls. So its ok to add a lock here if it makes the code simpler and safe.

Reply:
`synchronized` added.